src/mappings/handlers/blockHandlers.ts

```diff
@@ -65,12 +65,23 @@ async function _handleBlock(block: SubstrateBlock): Promise<void> {
       pool.resetUnrealizedProfit()
       for (const loanId in activeLoanData) {
         const asset = await AssetService.getById(pool.id, loanId)
-        const previousPrice = asset.currentPrice
+        const previousUnrealizedProfitByPeriod = asset.unrealizedProfitByPeriod
         await asset.updateActiveAssetData(activeLoanData[loanId])
+
+        const unrealizedProfitAtMarketPrice = await AssetPositionService.computeUnrealizedProfitAtPrice(
+          asset.id,
+          asset.currentPrice
+        )
+        const unrealizedProfitAtNotional = await AssetPositionService.computeUnrealizedProfitAtPrice(
+          asset.id,
+          asset.notional
+        )
+        logger.info(`byPeriod for ${asset.id} = ${unrealizedProfitAtMarketPrice} - ${previousUnrealizedProfitByPeriod}`)
+        const unrealizedProfitByPeriod = unrealizedProfitAtMarketPrice - previousUnrealizedProfitByPeriod
```

Comment:
can we define this logic of saving the previous value and subtracting it from the new one as part of the `AssetService.updateUnrealizedProfit()` method? Then all the detailed implementation resides in the service only and the LoansHandler is more readable ;)